pallets/automation-time/src/fees.rs

```diff
@@ -71,59 +78,110 @@ where
-		let fee = self.schedule_fee_amount.saturating_add(self.execution_fee_amount);
-
-		if fee.is_zero() {
+	fn ensure_can_withdraw(
+		&self,
+		asset_location: MultiLocation,
+		amount: MultiBalanceOf<T>,
+	) -> Result<(), DispatchError> {
+		if amount.is_zero() {
 			return Ok(())
 		}
 
-		// Manually check for ExistenceRequirement since MultiCurrency doesn't currently support it
-		let currency_id = T::CurrencyIdConvert::convert(self.schedule_fee_location)
-			.ok_or("IncoveribleMultilocation")?;
-		let currency_id = currency_id.into();
+		let currency_id = T::CurrencyIdConvert::convert(asset_location)
+			.ok_or("IncoveribleMultilocation")?
+			.into();
 		let free_balance = T::MultiCurrency::free_balance(currency_id, &self.owner);
+		let min_balance = T::MultiCurrency::minimum_balance(currency_id);
 
 		free_balance
-			.checked_sub(&fee)
-			.ok_or(DispatchError::Token(BelowMinimum))?
-			.checked_sub(&T::MultiCurrency::minimum_balance(currency_id))
+			.checked_sub(&amount)
+			.and_then(|balance_minus_fee| balance_minus_fee.checked_sub(&min_balance))
 			.ok_or(DispatchError::Token(BelowMinimum))?;
-		T::MultiCurrency::ensure_can_withdraw(currency_id, &self.owner, fee)?;
+
+		T::MultiCurrency::ensure_can_withdraw(currency_id, &self.owner, amount)?;
+
+		Ok(())
+	}
+
+	/// Ensure the fee can be paid.
+	fn can_pay_fee(&self) -> Result<(), DispatchError> {
+		match &self.execution_fee {
+			Some(exec_fee) if exec_fee.is_local => {
+				// If the locations of schedule_fee and execution_fee are equal,
+				// we need to add the fees to check whether they are sufficient,
+				// otherwise check them separately.
+				let exec_fee_location = exec_fee
+					.asset_location
+					.reanchored(&T::SelfLocation::get(), T::UniversalLocation::get())
+					.map_err(|_| Error::<T>::CannotReanchor)?;
+
+				let schedule_fee_location = self
+					.schedule_fee
+					.asset_location
+					.reanchored(&T::SelfLocation::get(), T::UniversalLocation::get())
+					.map_err(|_| Error::<T>::CannotReanchor)?;
+
+				if exec_fee_location == schedule_fee_location {
+					let fee = self.schedule_fee.amount.saturating_add(exec_fee.amount);
+					Self::ensure_can_withdraw(self, exec_fee.asset_location, fee)?;
+				} else {
+					Self::ensure_can_withdraw(
+						self,
+						self.schedule_fee.asset_location,
+						self.schedule_fee.amount,
+					)?;
+				}
+			},
+			_ => {
+				Self::ensure_can_withdraw(
+					self,
+					self.schedule_fee.asset_location,
+					self.schedule_fee.amount,
+				)?;
+			},
+		}
+
 		Ok(())
 	}
 
 	/// Withdraw the fee.
 	fn withdraw_fee(&self) -> Result<(), DispatchError> {
-		let fee = self.schedule_fee_amount.saturating_add(self.execution_fee_amount);
+		log::debug!(target: "FeeHandler", "FeeHandler::withdraw_fee, self.schedule_fee.asset_location: {:?}, self.schedule_fee.amount: {:?}",
+			self.schedule_fee.asset_location, self.schedule_fee.amount);
+		// Withdraw schedule fee
+		if !self.schedule_fee.amount.is_zero() {
```

Comment:
What happens if it is zero? A better structure would be
```
if is_zero() {
log debug or throw error
}
else{
the withdraw logic
}
```

Reply:
When the schedule_fee is not equal to zero, execute the withdrawal process; otherwise, skip it.

Reply:
Please add your explanation to this section of code as a comment.

```
When the expected deduction amount, schedule_fee_amount, is not equal to zero, execute the withdrawal process; otherwise, there’s no need to deduct.
```